### PR TITLE
Cleaned up BitcoinTxBuilder

### PR DESCRIPTION
**Pull request by nkohen**
Removed `UTXOMap` type alias and simplified construction of `BitcoinTxBuilder`

### CODE REVIEW

File: core/src/main/scala/org/bitcoins/core/wallet/builder/TxBuilder.scala [hunk 1]
```diff
@@ -74,25 +74,17 @@ sealed abstract class TxBuilder {
...
-  def utxoMap: TxBuilder.UTXOMap
-
-  def utxos: Seq[UTXOSpendingInfo] = utxoMap.values.toSeq
+  def utxos: Seq[UTXOSpendingInfo]
 
   /** This represents the rate, in [[org.bitcoins.core.wallet.fee.FeeUnit FeeUnit]], we
     * should pay for this transaction */
   def feeRate: FeeUnit
 
+  require(feeRate.toLong > 0L, "Specified fee was too low")
```
Comment (Christewart) [inline core/src/main/scala/org/bitcoins/core/wallet/builder/TxBuilder.scala]: This should probably just be added in the `FeeUnit` type as a invariant

File: wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala [hunk 2]
```diff
@@ -150,12 +149,10 @@ object Wallet extends WalletLogger {
...
-  def apply(
-      keyManager: BIP39KeyManager,
-      nodeApi: NodeApi,
-      chainQueryApi: ChainQueryApi)(
-      implicit config: WalletAppConfig,
-      ec: ExecutionContext): Wallet = {
+  def apply(keyManager: BIP39KeyManager,
+            nodeApi: NodeApi,
+            chainQueryApi: ChainQueryApi)(implicit config: WalletAppConfig,
+                                          ec: ExecutionContext): Wallet = {
```
Comment (Christewart) [inline wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala]: formatting issues?

Comment (nkohen) [reply wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala]: `scalafmt` did it :man_shrugging:

**Comment by Christewart** [comment]
CI fails because you didn't update documentation: https://travis-ci.org/bitcoin-s/bitcoin-s/jobs/632041725#L357

**Review by Christewart — APPROVED** [review]

**Pull request closed without merging**